tests/integration/resolve/test_issue_2568.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2024 Pex project contributors.
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import
+
+import re
+from textwrap import dedent
+
+from pex import targets
+from testing.cli import run_pex3
+from testing.pytest.tmp import Tempdir
+
+
+def test_independent_logs_for_independent_runs(tmpdir):
+    # type: (Tempdir) -> None
+
+    lock = tmpdir.join("lock.json")
+    log = tmpdir.join("pip.log")
+    run_pex3(
+        "lock", "sync", "--lock", lock, "--pip-log", log, "ansicolors==1.1.8", "cowsay==6.0"
```

Comment:
N.B.: This is the issue OP test, but with cowsay 6.0 instead of 6.1 so the Python 2.7 tests work.